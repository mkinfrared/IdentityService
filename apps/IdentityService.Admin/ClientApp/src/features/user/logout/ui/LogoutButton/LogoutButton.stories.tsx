import { Meta, StoryObj } from "@storybook/react";

import LogoutButton from "./LogoutButton";
import css from "./Story.module.scss";

type Story = StoryObj<typeof meta>;

const meta = {
  title: "UI/LogoutButton",
  component: LogoutButton,
  tags: ["autodocs"],
} satisfies Meta<typeof LogoutButton>;

const Default: Story = {
  args: {},
  render: (args) => (
    <div className={css.Story}>
      <LogoutButton {...args} />
    </div>
  ),
};

export { Default };

export default meta;
