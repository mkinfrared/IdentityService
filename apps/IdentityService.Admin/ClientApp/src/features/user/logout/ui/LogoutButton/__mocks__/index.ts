export { default } from "./LogoutButton.mock";
