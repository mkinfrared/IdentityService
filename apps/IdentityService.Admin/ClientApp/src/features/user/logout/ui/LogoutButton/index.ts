export { default } from "./LogoutButton";

export * from "./LogoutButton.type";
