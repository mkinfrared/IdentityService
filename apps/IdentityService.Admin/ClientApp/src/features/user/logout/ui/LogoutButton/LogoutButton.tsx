import { Button } from "@mui/material";

import { userManager } from "shared/lib/helpers";

import { LogoutButtonProps } from "./LogoutButton.type";

const LogoutButton = ({ className }: LogoutButtonProps) => {
  const logIn = () => {
    userManager.signoutRedirect();
  };

  return (
    <Button className={className} data-testid="LoginButton" onClick={logIn}>
      Logout
    </Button>
  );
};

export default LogoutButton;
