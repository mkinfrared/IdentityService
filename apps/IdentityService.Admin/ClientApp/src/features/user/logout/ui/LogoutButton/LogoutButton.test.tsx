import { render } from "@testing-library/react";

import LogoutButton from "./LogoutButton";

describe("<LogoutButton />", () => {
  const Component = <LogoutButton />;

  it("should be defined", () => {
    expect(LogoutButton).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });
});
