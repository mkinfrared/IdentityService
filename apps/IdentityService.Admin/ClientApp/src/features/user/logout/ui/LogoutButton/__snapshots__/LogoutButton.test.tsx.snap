// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoutButton /> should match the snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    data-testid="LoginButton"
    tabindex="0"
    type="button"
  >
    Logout
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
